rules/rbe_repo.bzl

```diff
@@ -274,66 +458,99 @@ def _rbe_autoconfig_impl(ctx):
-        config_repos.extend(ctx.attr.config_repos)
-    if config_repos:
-        # run the container and extract the autoconf directory
-        run_and_extract(
-            ctx,
-            bazel_version = ctx.attr.bazel_version,
-            bazel_rc_version = ctx.attr.bazel_rc_version,
-            config_repos = config_repos,
-            docker_tool_path = docker_tool_path,
-            image_name = image_name,
-            project_root = project_root,
-            use_default_project = use_default_project,
-        )
+    toolchain_config_spec_name = ctx.attr.toolchain_config_spec_name
+    if ctx.attr.config_version:
+        # If we found a config assing that to the toolchain_config_spec_name so when
+        # we produce platform BUILD file we can use it.
+        toolchain_config_spec_name = ctx.attr.config_version
+    else:
+        # If no config_version was found, generate configs
+        # Get the value of JAVA_HOME to set in the produced
+        # java_runtime
+        java_home = ctx.attr.java_home
+        if ctx.attr.create_java_configs:
+            java_home = get_java_home(ctx, docker_tool_path, image_name)
+            create_java_runtime(ctx, java_home)
+
+        config_repos = []
+        if ctx.attr.create_cc_configs:
+            config_repos.extend(_CONFIG_REPOS)
+        if ctx.attr.config_repos:
+            config_repos.extend(ctx.attr.config_repos)
+        if config_repos:
+            # run the container and extract the autoconf directory
+            run_and_extract(
+                ctx,
+                bazel_version = ctx.attr.bazel_version,
+                bazel_rc_version = ctx.attr.bazel_rc_version,
+                config_repos = config_repos,
+                docker_tool_path = docker_tool_path,
+                image_name = image_name,
+                project_root = mount_project_root,
+                use_default_project = use_default_project,
+            )
+
+        if ctx.attr.export_configs:
+            ctx.report_progress("expanding outputs")
 
-    ctx.report_progress("expanding outputs")
+            # If the user requested exporting configs and did not set a toolchain_config_spec_name lets pick the default
+            if not toolchain_config_spec_name:
+                toolchain_config_spec_name = ctx.attr.toolchain_config_suite_spec["default_toolchain_config_spec"]
 
-    # Expand outputs to project dir if user requested it
-    if ctx.attr.output_base:
-        expand_outputs(
+            # Create a default BUILD file with the platform + toolchain targets that
+            # will work with RBE with the produced toolchain (to be exported to
+            # output_dir)
+            ctx.report_progress("creating export platform")
+            create_export_platform(
+                ctx,
+                # Use "marketplace.gcr.io" instead of "l.gcr.io" in platform targets.
+                image_name = image_name.replace("l.gcr.io", "marketplace.gcr.io"),
```

Comment:
Need to do something similar for launcher.gcr.io? Also, are we sure this is always applicable for l.gcr.io links?

Reply:
This always applies for l.gcr.io / marketplace.gcr.io. No need to do it for launcher.gcr.io as this is just a trick to avoid pulling for the rbe ubuntu container which we define in default_toolchain_config_suite_spec in toolchain_config_suite_spec.bzl (i.e., for any other containers it will be fine to pull from marketplace or launcher).
In any case, leaving this as is is not a good idea, so moved logic to util.bzl and made it a bit narrower.